doc/specs/#2046 - Unified keybindings and commands, and synthesized action names.md

```diff
@@ -0,0 +1,589 @@
+---
+author: Mike Griese @zadjii-msft
+created on: 2020-06-15
+last updated: 2020-06-16
+issue id: 2046
+---
+
+# Command Palette, Addendum 1 -  Unified keybindings and commands, and synthesized action names
+
+## Abstract
+
+This document is intended to serve as an addition to the [Command Palette Spec].
+While that spec is complete in it's own right, subsequent discussion revealed
+additional ways to improve the functionality and usability of the command
+palette. This document builds largely on the topics already introduced in the
+original spec, so readers should first familiarize themselves with that
+document.
+
+One point of note from the original document was that the original specification
+was entirely too verbose when defining both keybindings and commands for
+actions. Consider, for instance, a user that wants to bind the action "duplicate
+the current pane". In that spec, they need to add both a keybinding and a
+command:
+
+```json
+{
+    "keybindings": [
+        { "keys": [ "ctrl+alt+t" ], "command": { "action": "splitPane", "split":"auto", "splitMode": "duplicate" } },
+    ],
+    "commands": [
+        { "name": "Duplicate Pane", "action": { "action": "splitPane", "split":"auto", "splitMode": "duplicate" }, "icon": null },
+    ]
+}
+```
+
+These two entries are practically the same, except for two key differentiators:
+* the keybinding has a `keys` property, indicating which key chord activates the
+  action.
+* The command has a `name` property, indicating what name to display for the
+  command in the Command Palette.
+
+What if the user didn't have to duplicate this action? What if the user could
+just add this action once, in their `keybindings` or `commands`, and have it
+work both as a keybinding AND a command?
+
+
+## Solution Design
+
+This spec will outline two primary changes to keybindings and commands.
+1. Unify keybindings and commands, so both `keybindings` and `commands` can
+   specify either actions bound to keys, and/or actions bound to entries in the
+   Command Palette.
+2. Propose a mechanism by which actions do not _require_ a `name` to appear in
+   the Command Palette.
+
+
+### Proposal 1: Unify Keybindings and Commands
+
+As noted above, keybindings and commands have nearly the exact same syntax, save
+for a couple properties. To make things easier for the user, I'm proposing
+treating everything in _both_ the `keybindings` _and_ the `commands` arrays as
+**BOTH** a keybinding and a command.
+* Anything entry that has a `keys` value will be added to the keybindings.
+  Pressing that keybinding will activate the action defined in `command`.

```

Comment:
QQ: How does this extend to pointer bindings?

Are these more `bindings` than `commands`?

Reply:
Oh, that's a good idea. `bindings` might be _most_ accurate.